fuel-txpool/src/txpool.rs

```diff
@@ -53,7 +54,7 @@ impl TxPool {
         }
         // check and insert dependency
         let rem = self.by_dependency.insert(&self.by_hash, db, &tx).await?;
-        self.by_hash.insert(tx.id(), tx.clone());
+        self.by_hash.insert(tx.id(), TxInfo::new(tx.clone()));
         self.by_gas_price.insert(&tx);
```

Comment:
Should we normalize these secondary indexes to just map to the tx hash instead of arc's?

Reply:
hm, maybe, i wrap it around `Arc` because i can clone it easily when i want to send list of transaction to block producer to use and i dont need to do deep copy, and second one is that it is used in multiple parts `by_pair` and  `by_hash`